### DISCUSSION TITLE
How does this differ to VirtualBox's seamless window management mode?

### DISCUSSION THREAD

**Discussion by roke-julian-lockhart** [winapps-org/winapps#295] [Support] [answered]
I refer to https://docs.oracle.com/en/virtualization/virtualbox/6.0/user/seamlesswindows.html#:~:text=4.6.-,Seamless%20Windows,-With%20the%20seamless:

<blockQuote>

 With the seamless windows feature of Oracle VM VirtualBox, you can have the windows that are displayed within a virtual machine appear side by side next to the windows of your host. This feature is supported for the following guest operating systems, provided that the Guest Additions are installed:

- Windows guests.
- Supported Linux or Oracle Solaris guests running the X Window System. 

After seamless windows are enabled, Oracle VM VirtualBox suppresses the display of the desktop background of your guest, allowing you to run the windows of your guest operating system seamlessly next to the windows of your host.

![Figure 4.2 Seamless Windows on a Host Desktop](https://docs.oracle.com/en/virtualization/virtualbox/6.0/user/images/seamless.png)

</blockQuote>

I'm struggling to ascertain what the purpose of this is, unless, perhaps, the aforereferenced functionality is unavailable on the developers' configurations? [^1]

[^1]: https://www.reddit.com/r/linux4noobs/comments/6jzb31/comment/dji5kdm/#:~:text=Used%20it%20for%20a%20bit%20because%20i%20needed%20to%20make%20a%20presentation%20in%20Powerpoint.%20It%20worked%20seamlessly%20:D

**Comment by gordoncheong** [comment, marked as answer]
1. VirtualBox is proprietary. Someone people might have issues with that.
2. VirtualBox is a type-2 hypervisor while we leverage KVM, which is a type-1 hypervisor and much more performant.
3. [RAIL](https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-rdperp/3aa1de2c-6353-4cc8-b33a-8907ca386c67) in the RDP protocol is used instead of cutting/suppressing display space.
4. System icon support, which makes the program acts much more like a native application.
5. Mimetype and RDP file pass-through support, which allows you to open files directly on /home and also makes the program acts much more like a native application.

**Reply by roke-julian-lockhart** [reply]
> https://github.com/winapps-org/winapps/discussions/295#discussioncomment-11001553

@gordoncheong, so in practice, it's rather more like a counterpart to WayDroid? Irrespective, I am thankful for such a comprehensive but concise answer.

**Reply by oskardotglobal** [reply]
Yeah, that's about right, or at least we try to be
At least WayDroid properly supports Wayland....
